src/codec_aom.c

```diff
@@ -1032,41 +1029,45 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
     }
 
     unsigned char * monoUVPlane = NULL;
-    if (monochromeRequested && !codec->internal->monochromeEnabled) {
-        // The user requested monochrome (via alpha or YUV400) but libaom cannot currently support
-        // monochrome (see chroma_check comment above). Manually set UV planes to 0.5.
-
-        // aomImage is always 420 when we're monochrome
-        uint32_t monoUVWidth = (image->width + 1) >> 1;
-        uint32_t monoUVHeight = (image->height + 1) >> 1;
-
-        // Allocate the U plane if necessary.
-        if (!aomImageAllocated) {
-            uint32_t channelSize = avifImageUsesU16(image) ? 2 : 1;
-            uint32_t monoUVRowBytes = channelSize * monoUVWidth;
-            size_t monoUVSize = (size_t)monoUVHeight * monoUVRowBytes;
-
-            monoUVPlane = avifAlloc(monoUVSize);
-            aomImage.planes[1] = monoUVPlane;
-            aomImage.stride[1] = monoUVRowBytes;
-        }
-        // Set the U plane to 0.5.
-        if (image->depth > 8) {
-            const uint16_t half = 1 << (image->depth - 1);
-            for (uint32_t j = 0; j < monoUVHeight; ++j) {
-                uint16_t * dstRow = (uint16_t *)&aomImage.planes[1][(size_t)j * aomImage.stride[1]];
-                for (uint32_t i = 0; i < monoUVWidth; ++i) {
-                    dstRow[i] = half;
+    if (monochromeRequested) {
+        if (codec->internal->monochromeEnabled) {
+            aomImage.monochrome = 1;
```

Comment:
I think this is a good change, but it turns out that the libaom encoder ignores this field (and also the fields we set on lines 1024-1028). What tells the libaom encoder that the image is monochrome is line 730:

```
            if (alpha || (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV400)) {
                codec->internal->monochromeEnabled = AVIF_TRUE;
                cfg->monochrome = 1;  <=== THIS IS IMPORTANT
            }
```